viewer_clients/web-client/scripts/codecheckerviewer/BugViewer.js

```diff
@@ -462,49 +462,204 @@ function (declare, dom, style, on, query, Memory, Observable, topic, entities,
+
+          highlight.iconOverride = "calling";
+        } else if (startsWith("Entered call from ")) {
+          highlight.iconOverride = "entered_call";
+        } else if (func = extractFuncName("Returning from ")) {
+          if (func === stack.funcStack[stack.funcStack.length - 1]) {
+            stack.funcStack.pop();
+            stack.background = that._highlight_colours[stack.funcStack.length % that._highlight_colours.length];
+
+            highlight.iconOverride = "returning";
+          } else {
+            throw {
+              name : 'StackError',
+              message : "Returned from " + func + " while the last function was " +
+                        stack.funcStack[stack.funcStack.length - 1]
+            };
+          }
+        } else if (startsWith("Assuming the condition")) {
+          highlight.iconOverride = "assume_switch";
+        } else if (startsWith("Assuming")) {
+          highlight.iconOverride = "assume_exclamation";
+        } else if (msg == "Entering loop body") {
+          highlight.iconOverride = "loop_enter";
+        } else if (startsWith("Loop body executed")) {
+          highlight.iconOverride = "loop_execute";
+        } else if (msg == "Looping back to the head of the loop") {
+          highlight.iconOverride = "loop_back";
+        }
+      };
+
+      stackTransition();
+
+      return highlight;
+    },
+
     _addReport : function (report) {
       var that = this;
 
       this.bugStore.put({
         id : report.reportId + '',
         name : 'Line ' + report.lastBugPosition.startLine
-             + ': ' + report.checkerId,
+             + ' &ndash; ' + report.checkerId,
         parent : util.severityFromCodeToString(report.severity),
         report : report,
-        isLeaf : false
+        isLeaf : false,
+        kind : 'bugpath'
       });
 
       this.bugStore.put({
         id : report.reportId + '_0',
-        name : '<b><u>Result</u>: '
-             + entities.encode(report.checkerMsg)
-             + '</b>',
+        name : '<b><u>' + entities.encode(report.checkerMsg) + '</u></b>',
         parent : report.reportId + '',
         report : report,
-        isLeaf : true
+        isLeaf : true,
+        kind : 'msg'
       });
 
+      var filePaths = [];
+      var areThereMultipleFiles = false;
+      var highlightStack = {
+        funcStack : [],
+        background : this._highlight_colours[0]
+      };
+
       CC_SERVICE.getReportDetails(report.reportId, function (reportDetails) {
-        reportDetails.pathEvents.forEach(function (step, index) {
-          that.bugStore.put({
-            id : report.reportId + '_' + (index + 1),
-            name : 'Line ' + step.startLine + ': ' + entities.encode(step.msg),
-            parent : report.reportId,
-            bugPathEvent : step,
-            isLeaf : true,
-            report : report
+        if (reportDetails.pathEvents.length > 1) {
```

Comment:
You can reduce an indentation level if you negate this condition and simply return from this callback.